curricula/models.py

```diff
@@ -454,46 +458,31 @@ def jackfrost_urls(self):
     def pdf_urls(self):
         return [self.get_pdf_url(), self.get_resources_pdf_url()]
 
-    def jackfrost_can_build(self):
-        return settings.ENABLE_PUBLISH and self.status == 2 and not self.login_required and not self.curriculum.login_required
+    def yield_urls_content(self, urls, slack_message_prefix, silent):
+        if self.jackfrost_can_build():
+            for url in urls:
+                read, written = build_single(url)
+                if not silent:
+                    slack_message('slack/message.slack', {
+                        'message': '%s %s %s %s' % (slack_message_prefix, self.content_model, self.title, url),
+                        'color': '#00adbc'
+                    })
+                yield json.dumps(written)
+                yield '\n'
+        else:
+            raise RuntimeError('Unable to generate content: jackfrost cannot build')
```

Comment:
How does this get surfaced back to the user?

Reply:
It bubbles back up to the publish* methods, which in turn bubble up to the publish method, which handles the exception.

In general it seems best to handle exceptions as late as possible, to remove the need for redundant exception logging code.

Reply:
Will the user get something in the UI that tells them what the problem is comprehensibly?

Reply:
I just spent some time looking into this more closely. The answer to your question appears to be 'no' - also, any exception thrown by the publish* functions look like they won't inform the user of a problem.

The publish* functions are generator, and generators in python handle exceptions differently than other functions in python - the exception isn't seen until after the StreamingHttpResponse is returned by the view and the user has gotten HTTP headers returned to them

Reply:
Adding exception logging into the generator functions would still log the issue to our server, but as-is I think It'd look like a silent failure to users